### PR TITLE
Unify DevSupportManager and WebSocketJSExecutor

### LINKED ISSUE #4985: Unify DevSupportManagerUWP and DevSupportManager
they're pretty much the same code (in fact there have been copies of pieces of code shuttled back and forth apparently), we should just converge them

### PR DESCRIPTION
**Pull request by acoates-ms**
Desktop and UWP had two copies of DevSupportManager and WebSocketJSExecutor.  The only reason for this was Desktop was supporting win7 and so couldn't use the winrt APIs for http/websocket.  Now that we no longer support win7 on desktop these implementations can be unified.

On top of that, the desktop version had not been updated to support 0.62, which changes the dev support protocol to use a websocket instead http - so reload scenarios wouldn't work on desktop.

This change switches desktop to use the UWP implementation of DevSupportManager and WebSocketJSExecutor.

Fixes #4985 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/5231)

### CODE REVIEW

File: vnext/Desktop.IntegrationTests/WebSocketJSExecutorIntegrationTest.cpp [hunk 1]
```diff
@@ -86,14 +91,15 @@ TEST_CLASS (WebSocketJSExecutorIntegrationTest) {
...
+    auto noop = []() {};
+    jse->ConnectAsync("ws://localhost:8081/debugger-proxy?role=client", move(errorCallback), noop, noop).get();
     // Point to a non-existing path.
     auto bigString = unique_ptr<JSBigString>(new JSBigStdString("http://localhost:8081/showme404"));
     jse->loadApplicationScript(std::move(bigString), "");
 
     jsThread->quitSynchronous();
 
-    Assert::IsTrue(connected);
+    // Assert::IsTrue(connected);
```
Comment (TheSavior) [inline vnext/Desktop.IntegrationTests/WebSocketJSExecutorIntegrationTest.cpp:102]: Do you check in commented code?

Comment (acoates-ms) [reply vnext/Desktop.IntegrationTests/WebSocketJSExecutorIntegrationTest.cpp]: No we don't.  I deleted the whole file instead. (They were dead tests)

File: .ado/windows-vs-pr.yml [hunk 2]
```diff
@@ -374,7 +374,7 @@ jobs:
 
     variables:
       #5059 Disable failing or intermittent tests.
-      Desktop.IntegrationTests.Filter: '(FullyQualifiedName!~WebSocketJSExecutorIntegrationTest)&(FullyQualifiedName!~WebSocket)&(FullyQualifiedName!=RNTesterIntegrationTests::IntegrationTestHarness)&(FullyQualifiedName!=RNTesterIntegrationTests::AsyncStorage)'
+      Desktop.IntegrationTests.Filter: '(FullyQualifiedName!~WebSocket)&(FullyQualifiedName!=RNTesterIntegrationTests::IntegrationTestHarness)&(FullyQualifiedName!=RNTesterIntegrationTests::AsyncStorage)'
```
Comment (JunielKatarn) [inline .ado/windows-vs-pr.yml:377]: Why were all WebSocket tests disabled?

Comment (JunielKatarn) [reply .ado/windows-vs-pr.yml:377]: Misread the entry.
The excluded tests were already an issue.

**Comment by Khalef1** [comment]
@JunielKatarn : Does this still need a backport to 60 and 61?

**Comment by JunielKatarn** [comment]
> @JunielKatarn : Does this still need a backport to 60 and 61?

We haven't seen recent issues hitting this component.
As Office's 0.62 upgrade approaches, I would say no.

**Comment by Khalef1** [comment]
Ok, removing the backport request tags then

**Pull request merged**